The branch is having some problems in the build system.

Linux 'release build':

```
Installing 32/33 speexdsp:x64-linux@1.2.1#1...
speexdsp:x64-linux@1.2.1#1 package ABI: d72ddbd72fa9eb098d63da45a5031ce7084ed3b561bcfd5ce2a95546e620b4ca
Building speexdsp:x64-linux@1.2.1#1...
Downloading http://downloads.xiph.org/releases/speex/speexdsp-1.2.1.tar.gz -> speexdsp-1.2.1.tar.gz
Attempt 1 of 3, retrying download.
Attempt 2 of 3, retrying download.
error: Download timed out.
error: Download timed out.
error: Reached maximum number of attempts, won't retry download from http://downloads.xiph.org/releases/speex/speexdsp-1.2.1.tar.gz.
```

Windows 'generate release notes':

```
Downloading urllib3-2.6.3-py3-none-any.whl (131 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 131.6/131.6 kB 3.0 MB/s eta 0:00:00
Installing collected packages: urllib3, markdown2, idna, charset_normalizer, certifi, requests
Successfully installed certifi-2026.4.22 charset_normalizer-3.4.7 idna-3.13 markdown2-2.5.5 requests-2.33.1 urllib3-2.6.3
Run ./scripts/tools/release-notes-drafter.py \
Traceback (most recent call last):
Fetching next page of pull requests, item offset: 0
  File "/home/runner/.local/lib/python3.12/site-packages/urllib3/connectionpool.py", line 534, in _make_request
Fetching next page of pull requests, item offset: 50
    response = conn.getresponse()
               ^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/lib/python3.12/site-packages/urllib3/connection.py", line 571, in getresponse
    httplib_response = super().getresponse()
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/http/client.py", line 1448, in getresponse
Fetching next page of pull requests, item offset: 100
Fetching next page of pull requests, item offset: 150
Fetching next page of pull requests, item offset: 200
Fetching next page of pull requests, item offset: 250
Fetching next page of pull requests, item offset: 300
    response.begin()
  File "/usr/lib/python3.12/http/client.py", line 336, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/http/client.py", line 297, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/socket.py", line 707, in readinto
    return self._sock.recv_into(b)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/ssl.py", line 1252, in recv_into
    return self.read(nbytes, buffer)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/ssl.py", line 1104, in read
    return self._sslobj.read(len, buffer)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TimeoutError: The read operation timed out
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,20 @@
 cmake_minimum_required(VERSION 3.25)
 
+# The project only tests preset build configurations, so we use
+# the IS_PRESET_USED variable to check and recommend using a preset.
+
+if (NOT IS_PRESET_USED)
+  message(WARNING
+    "\nPer docs/build-*.md, we recommend using a platform-specific preset.
+     cmake --preset <name>"
+  )
+  execute_process(
+    COMMAND ${CMAKE_COMMAND} --list-presets
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  )
+  message("")
+endif()
+
 set(VCPKG_USE_HOST_TOOLS ON)
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -475,8 +475,15 @@ set(LICENSE_FILES
   licenses/Zlib.txt
 )
 
-# Set the directory with resources to bundle
+# The source of the resources, relative to the source root
 set(RESOURCES_PATH "resources")
+
+# The copy target of the resources, relative to the compiled binary
+if (APPLE OR WIN32)
+  set(RESOURCE_COPY_PATH "Resources" CACHE STRING "Resources copy target relative to binary")
+else()
+  set(RESOURCE_COPY_PATH "resources" CACHE STRING "Resources copy target relative to binary")
+endif()
 
 # Generate asset copy commands and add them as a dependency
 include(cmake/add_copy_assets.cmake)

--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -8,6 +8,7 @@
       "generator": "Ninja",
       "binaryDir": "${sourceDir}/build/debug",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
         "CMAKE_BUILD_TYPE": "Debug"
@@ -20,6 +21,7 @@
       "generator": "Ninja",
       "binaryDir": "${sourceDir}/build/release",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
         "CMAKE_BUILD_TYPE": "Release"
@@ -32,6 +34,7 @@
       "generator": "Ninja",
       "binaryDir": "${sourceDir}/build/debug-linux",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_BUILD_TYPE": "Debug",
         "USE_SYSTEM_LIBS": "ON"
       }
@@ -43,6 +46,7 @@
       "generator": "Ninja",
       "binaryDir": "${sourceDir}/build/release-linux",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_BUILD_TYPE": "Release",
         "USE_SYSTEM_LIBS": "ON"
       }
@@ -54,6 +58,7 @@
       "generator": "Ninja",
       "binaryDir": "${sourceDir}/build/debug-linux",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
         "CMAKE_BUILD_TYPE": "Debug"
@@ -66,6 +71,7 @@
       "generator": "Ninja",
       "binaryDir": "${sourceDir}/build/release-linux",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
         "CMAKE_BUILD_TYPE": "Release"
@@ -78,6 +84,7 @@
       "generator": "Xcode",
       "binaryDir": "${sourceDir}/build/debug-macos",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
@@ -91,6 +98,7 @@
       "generator": "Xcode",
       "binaryDir": "${sourceDir}/build/release-macos",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
@@ -104,6 +112,7 @@
       "generator": "Xcode",
       "binaryDir": "${sourceDir}/build/release-macos-arm64",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_OSX_ARCHITECTURES": "arm64",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
@@ -118,6 +127,7 @@
       "generator": "Xcode",
       "binaryDir": "${sourceDir}/build/release-macos-x86_64",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_OSX_ARCHITECTURES": "x86_64",
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
@@ -132,6 +142,7 @@
       "generator": "Visual Studio 17 2022",
       "binaryDir": "${sourceDir}/build/debug-windows",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_GENERATOR_TOOLSET": "ClangCL",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
@@ -149,6 +160,7 @@
       "generator": "Visual Studio 17 2022",
       "binaryDir": "${sourceDir}/build/release-windows",
       "cacheVariables": {
+        "IS_PRESET_USED" : "TRUE",
         "CMAKE_GENERATOR_TOOLSET": "ClangCL",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",

--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -10,8 +10,7 @@
       "cacheVariables": {
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Debug",
-        "RESOURCE_COPY_PATH": "/../resources"
+        "CMAKE_BUILD_TYPE": "Debug"
       }
     },
     {
@@ -23,8 +22,7 @@
       "cacheVariables": {
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Release",
-        "RESOURCE_COPY_PATH": "/../resources"
+        "CMAKE_BUILD_TYPE": "Release"
       }
     },
     {
@@ -35,7 +33,6 @@
       "binaryDir": "${sourceDir}/build/debug-linux",
       "cacheVariables": {
         "CMAKE_BUILD_TYPE": "Debug",
-        "RESOURCE_COPY_PATH": "/resources",
         "USE_SYSTEM_LIBS": "ON"
       }
     },
@@ -47,7 +44,6 @@
       "binaryDir": "${sourceDir}/build/release-linux",
       "cacheVariables": {
         "CMAKE_BUILD_TYPE": "Release",
-        "RESOURCE_COPY_PATH": "/resources",
         "USE_SYSTEM_LIBS": "ON"
       }
     },
@@ -60,8 +56,7 @@
       "cacheVariables": {
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Debug",
-        "RESOURCE_COPY_PATH": "/resources"
+        "CMAKE_BUILD_TYPE": "Debug"
       }
     },
     {
@@ -73,8 +68,7 @@
       "cacheVariables": {
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Release",
-        "RESOURCE_COPY_PATH": "/resources"
+        "CMAKE_BUILD_TYPE": "Release"
       }
     },
     {
@@ -87,8 +81,7 @@
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Debug",
-        "RESOURCE_COPY_PATH": "/Resources"
+        "CMAKE_BUILD_TYPE": "Debug"
       }
     },
     {
@@ -101,8 +94,7 @@
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Release",
-        "RESOURCE_COPY_PATH": "/Resources"
+        "CMAKE_BUILD_TYPE": "Release"
       }
     },
     {
@@ -116,8 +108,7 @@
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Release",
-        "RESOURCE_COPY_PATH": "/Resources"
+        "CMAKE_BUILD_TYPE": "Release"
       }
     },
     {
@@ -131,8 +122,7 @@
         "CMAKE_OSX_DEPLOYMENT_TARGET": "12.0",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Release",
-        "RESOURCE_COPY_PATH": "/Resources"
+        "CMAKE_BUILD_TYPE": "Release"
       }
     },
     {
@@ -145,8 +135,7 @@
         "CMAKE_GENERATOR_TOOLSET": "ClangCL",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Debug",
-        "RESOURCE_COPY_PATH": "/Resources"
+        "CMAKE_BUILD_TYPE": "Debug"
       },
       "environment": {
         "UseMultiToolTask": "true",
@@ -163,8 +152,7 @@
         "CMAKE_GENERATOR_TOOLSET": "ClangCL",
         "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
         "CMAKE_FIND_PACKAGE_PREFER_CONFIG" : "ON",
-        "CMAKE_BUILD_TYPE": "Release",
-        "RESOURCE_COPY_PATH": "/Resources"
+        "CMAKE_BUILD_TYPE": "Release"
       },
       "environment": {
         "UseMultiToolTask": "true",

--- a/cmake/add_build_documentation.cmake
+++ b/cmake/add_build_documentation.cmake
@@ -163,7 +163,7 @@ function(add_build_documentation)
   set(MKDOCS_VENV_DIR      "${CMAKE_BINARY_DIR}/_mkdocs_venv")
   set(MKDOCS_REQUIREMENTS  "${CMAKE_SOURCE_DIR}/extras/documentation/mkdocs-package-requirements.txt")
   set(MKDOCS_CONFIG        "${CMAKE_SOURCE_DIR}/website/mkdocs.yml")
-  set(MKDOCS_OUTPUT_DIR    "${CMAKE_CURRENT_BINARY_DIR}${RESOURCE_COPY_PATH}/docs")
+  set(MKDOCS_OUTPUT_DIR    "${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE_COPY_PATH}/docs")
   set(MKDOCS_PIP_STAMP     "${MKDOCS_VENV_DIR}/.pip_install_stamp")
   set(MKDOCS_BUILD_STAMP   "${CMAKE_BINARY_DIR}/_mkdocs_build_stamp")
 

--- a/cmake/add_copy_assets.cmake
+++ b/cmake/add_copy_assets.cmake
@@ -22,7 +22,7 @@ function(add_copy_assets)
   # Generate list of resource files with destination paths
   foreach(RESOURCE_FILE ${RESOURCE_FILES})
     list(APPEND RESOURCE_DESTINATION_FILES
-         "${CMAKE_CURRENT_BINARY_DIR}${RESOURCE_COPY_PATH}/${RESOURCE_FILE}")
+         "${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE_COPY_PATH}/${RESOURCE_FILE}")
   endforeach()
 
   # Generate list of license files with destination paths
@@ -41,8 +41,8 @@ function(add_copy_assets)
   # Add a copy command for each resource file
   foreach(RESOURCE_FILE ${RESOURCE_FILES})
     add_to_copy_command(
-      "${CMAKE_CURRENT_SOURCE_DIR}/resources/${RESOURCE_FILE}"
-      "${CMAKE_CURRENT_BINARY_DIR}${RESOURCE_COPY_PATH}/${RESOURCE_FILE}"
+      "${CMAKE_CURRENT_SOURCE_DIR}/${RESOURCES_PATH}/${RESOURCE_FILE}"
+      "${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE_COPY_PATH}/${RESOURCE_FILE}"
     )
   endforeach()
 

--- a/cmake/add_install_rules.cmake
+++ b/cmake/add_install_rules.cmake
@@ -41,7 +41,7 @@ function(add_install_rules)
     endforeach()
 
     # Bundle the resources
-    install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}${RESOURCE_COPY_PATH}/"
+    install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE_COPY_PATH}"
             DESTINATION "${INSTALL_DIR_RESOURCES}")
 
     # Bundle required licenses

--- a/docs/build-linux.md
+++ b/docs/build-linux.md
@@ -4,8 +4,12 @@ Two build methods are available for Linux — using the [system libraries](#buil
 provided by your Linux distribution or using the [vcpkg tool](#building-using-vcpkg)
 to fetch and compile dependencies.
 
-The vcpkg method is used by the team to provide our official binaries, which are
-intended to be run on different distros — they only depend on glibc.
+Both library options are available using CMake presets, which we highly
+recommend because they're CI-tested and produce a binary using consistent
+compiler flags. Run `cmake --list-presets` to list the presets.
+
+The vcpkg presets are used by the team to provide our official binaries, which
+are intended to be run on different distros — they only depend on glibc.
 
 ## Packaging
 

--- a/docs/build-macos.md
+++ b/docs/build-macos.md
@@ -1,8 +1,12 @@
 # Building on macOS
 
-macOS builds can be created using the Meson buildsystem, compiled using the
-Clang or GCC compilers, and provided with dependencies using the Homebrew or
-MacPorts package managers.
+macOS builds can be created using CMake or the Meson build tools, compiled
+using the Clang or GCC compilers, and provided with dependencies using the
+Homebrew or MacPorts package managers.
+
+We recommend using CMake with presets because they're CI-tested and produce a
+binary using consistent compiler flags. Run `cmake --list-presets` to list the
+presets.
 
 We recommend using Homebrew and Clang because Apple's Core SDKs can be used
 only with Apple's fork of the Clang compiler.

--- a/docs/build-windows.md
+++ b/docs/build-windows.md
@@ -1,6 +1,11 @@
 # Building on Windows
 
-Windows builds can be created using:
+Windows builds can be created using CMake, Visual Studio, or Meson:
+
+- CMake is used to produce the official release package, and therefore is the
+  recommended build tool. We also recommend using presets because they're
+  CI-tested and produce a binary using consistent compiler flags. Run `cmake
+  --list-presets` to list the presets.
 
 - Visual Studio 2022 IDE suite with Clang/LLVM compiler, and vcpkg to provide
   dependencies. This is the fully-supported toolchain used to create release
@@ -9,13 +14,6 @@ Windows builds can be created using:
 - The Clang or GCC compilers using the Meson build system running within the
   MSYS2 environment to provide dependencies (deprecated, support will be probably
   removed in the future).
-
-> **Note**
->
-> CMake support is currently an experimental internal-only, work-in-progress
-> feature; it's not ready for public consumption yet. Please ignore the
-> `CMakeLists.txt` files in the source tree.
-
 
 ## Build using Visual Studio
 

